src/shell/xdg/window/mod.rs

```diff
@@ -208,8 +208,8 @@ impl Window {
         decoration.data::<WindowData>().and_then(|data| data.0.upgrade()).map(Window)
     }
 
-    pub fn show_window_menu(&self, seat: &wl_seat::WlSeat, serial: u32, position: (u32, u32)) {
-        self.xdg_toplevel().show_window_menu(seat, serial, position.0 as i32, position.1 as i32);
+    pub fn show_window_menu(&self, seat: &wl_seat::WlSeat, serial: u32, position: (i32, i32)) {
```

Comment:
That was wrong, it's i32, because relative to main surface, so could be negative.